test/support/test_config.h

```diff
@@ -144,11 +144,17 @@
 #define _PSTL_TEST_COMPLEX_OP_BROKEN (_MSVC_STL_VERSION && _MSVC_STL_VERSION <= 143)
 
 #define _PSTL_TEST_COMPLEX_ACOS_BROKEN _PSTL_TEST_COMPLEX_OP_BROKEN
+#define _PSTL_TEST_COMPLEX_ACOS_BROKEN_GLIBCXX                                                                         \
+    (__SYCL_DEVICE_ONLY__ && __INTEL_LLVM_COMPILER > 0 && __INTEL_LLVM_COMPILER <= 20240200 && __GLIBCXX__ > 0 &&      \

```

Comment:
I've not seen us protect this with `__SYCL_DEVICE_ONLY__` before.  Is this necessary?
If so, should we be doing this for many more of these macros?